src/components/view/ContactTypeaheadSearch/index.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+
+import AutoComplete from 'material-ui/AutoComplete';
+import CancelIcon from 'material-ui/svg-icons/navigation/cancel';
+import Paper from 'material-ui/Paper';
+
+import muiThemeable from 'material-ui/styles/muiThemeable';
+
+class ContactTypeaheadSearch extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    handleUpdateInput = (value) => {
+        this.validate(value);
+
+        this.props.typeaheadSearch(value);
+
+        // navigate to contact search results page if we're not already there
+        if (this.props.location.pathname !== '/contact') {
+            this.props.history.push('/contact');
+        }
+    };
+
+    validate(value) {
+        let regex;
+        let errorMsg = '';
+        // can this be done with a single RegEx??
+        if (value.length > 0 && value.length <= 4) {
+            regex = new RegExp('(\\w{1,4})');
+        }
+        else if (value.length > 4 && value.length <= 10) {
+            regex = new RegExp('\\w{4}\\d{1,6}');
+        }
+        else if (value.length > 10) {
```

Comment:
Bump `else` up to line with `{`
#optional

Reply:
okay, I've added a `.eslintrc` file and put that rule in there so that we'll see output in our editors and will keep the style consistent. 

Looks like one downside of create react app is that you cannot extend their eslint ruleset and have those extended rules get picked up and displayed in the browser or terminal 
https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#displaying-lint-output-in-the-editor

But at least this way we'll catch it in the editor more easily